exercises/bowling/example.rs

```diff
@@ -96,20 +102,20 @@ impl BowlingGame {
         BowlingGame { frames: vec![Frame::new()] }
     }
 
-    pub fn roll(&mut self, pins: u16) -> Result<(), &'static str> {
+    pub fn roll(&mut self, pins: u16) -> Result<(), Error> {
         if pins > 10 {
-            Err("Greater than 10 pins")
+            Err(Error::NotEnoughPinsLeft)
```

Comment:
discussion: I do not favour making a separate error variant for `GreaterThanTenPins` since I don't really find it useful to distinguish between a roll that is always invalid and a roll that is sometimes valid but currently invalid given the game state. What does the caller do differently if it were able to distinguish this information?

Reply:
Makes sense to me.